Mailutils v3.4 Fails to Build and Install Under OS X v10.11.6 'El Capitan'

Poster's Note:  

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Apologies for the delay; I hit a little snag in the form of the cause of [this Stack Exchange question](https://unix.stackexchange.com/questions/401956/how-do-i-unbuffer-the-output-passed-from-an-interactive-command-into-a-pipeline) while trying to gather logs for this issue.  _Any_-way, here's what I actually _meant_ to file when I accidentally submitted #20154.  Also, this issue is now different than the one I was _originally_ going to submit now that Mailutils v3.4 is out and Homebrew has registered that update's existence, but this same exact bug arose with Mailutils v3.3 on my machine when _that_ was still the latest version of the software available through Homebrew.  

<hr />

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

# Please always follow these steps:
- [x] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [ ] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why:)  
   
   Upgrade my local, built-from-source copy of Mailutils from v3.2 to v3.4.  

- What happened (include command output:)  
   
   Homebrew failed to successfully build Mailutils v3.4 from source and install it in place of v3.2.  My logs are:  

   - [here](https://gitlab.com/RandomDSdevel/Homebrew-Installation-Logs/tree/master/Mailutils/v3.4/Recorded%20Attempts/First%20Recorded%20Attempt/With%20%60HOMEBREW_BUILD_FROM_SOURCE=1%60) (from when I _did_ set `HOMEBREW_BUILD_FROM_SOURCE=1`.)  
   - [here](https://gitlab.com/RandomDSdevel/Homebrew-Installation-Logs/tree/master/Mailutils/v3.4/Recorded%20Attempts/First%20Recorded%20Attempt/With%20%60HOMEBREW_BUILD_FROM_SOURCE%60%20Undefined) (from when I _didn't._)  
   
   In both cases, the exact point of failure appeared as follows:  
   
   ```
   ⋮
   
     CC       mu_body.lo
   mu_body.c:43:8: error: unknown type name 'scm_sizet__GONE__REPLACE_WITH__size_t'
   static scm_sizet
          ^
   /usr/local/Cellar/guile/2.2.2/include/guile/2.2/libguile/deprecated.h:121:19: note: expanded from macro 'scm_sizet'
   #define scm_sizet scm_sizet__GONE__REPLACE_WITH__size_t
                     ^
   1 error generated.
   make[3]: *** [mu_body.lo] Error 1
   make[3]: *** Waiting for unfinished jobs....
   make[2]: *** [install-recursive] Error 1
   make[1]: *** [install] Error 2
   make: *** [install-recursive] Error 1
   /usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python.rb
   /usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
   BuildError: Failed executing: make PYTHON_LIBS=-undefined\ dynamic_lookup install
   1. raise
   2. ignore
   3. backtrace
   4. irb
   5. shell
   Choose an action: …
   ⋮
   ```
   
   From what I can see, this looks like it could be either a bug in GUILE v2.2.2, which is what I have installed, or an API change in GUILE which has not yet been reflected in Mailutils to me.  I'll be happy to report this to the relevant upstream developers if you aren't.  
   
   (This issue also applies to Mailutils v3.3, thus hopefully explaining why I hadn't installed it yet as of my submission of this issue:  I would have if I hadn't been trying to put together a version of this issue for _that_ version of the software.)  

- What you expected to happen:  
   
   For Homebrew to successfully build the new version of Mailutils from source and install it in the old version's place.  

- Step-by-step reproduction instructions (by running `brew` commands:)  
   - _With_ `HOMEBREW_BUILD_FROM_SOURCE=1` (not supported with issue resolution:)  
      
      ```
      HOMEBREW_BUILD_FROM_SOURCE=1 brew upgrade -vd --build-from-source mailutils
      ```
      
      Note that setting that environment variable as shown above _probably_ doesn't make _any_ difference in this case, as upgrading Mailutils doesn't depend on upgrading any of its dependencies given the states my Homebrew installation and the packages it manages are currently in, but I usually set it _any_ time I run `brew upgrade`, just to be consistent.  

   - _Without_ `HOMEBREW_BUILD_FROM_SOURCE=1` (given to show that this issue _also_ occurs under a _supported_ configuration:)  
   
      ```
      brew upgrade -vd --build-from-source mailutils
      ```

--- a/Formula/mailutils.rb
+++ b/Formula/mailutils.rb
@@ -19,6 +19,7 @@ class Mailutils < Formula
   def install
     system "./configure", "--disable-mh",
                           "--prefix=#{prefix}",
+                          "--without-guile",
                           "--without-tokyocabinet"
     system "make", "PYTHON_LIBS=-undefined dynamic_lookup", "install"
   end

